flink-connectors/flink-connector-base/src/main/java/org/apache/flink/connector/base/source/reader/SourceReaderBase.java

```diff
@@ -114,16 +137,20 @@ public InputStatus pollNext(ReaderOutput<T> output) throws Exception {
 		if (recordsWithSplitId == null) {
 			recordsWithSplitId = getNextFetch(output);
 			if (recordsWithSplitId == null) {
+				idleSince = clock.absoluteTimeMillis();
```

Comment:
For measuring intervals, relative time (which is stable) seems like a good choice.